### PR TITLE
Added DenseNet

### PR DESCRIPTION
**Pull request by titu1994**
DenseNet in densenet.py (in applications folder) allows to construct arbitrary DenseNet models.

Weights provided for DenseNet-40-12 for CIFAR 10.

### CODE REVIEW

**Comment by titu1994** [comment]
Sorry about multiple commits, couldn't figure out how to force PyCharm to show PEP8 violations as errors. Fixed now.

File: keras_contrib/applications/densenet.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                model.load_weights(weights_path)
+
+                if K.backend() == 'theano':
+                    convert_all_kernels_in_model(model)
+
+    return model
+
+
+def __conv_block(ip, nb_filter, bottleneck=False, dropout_rate=None, weight_decay=1E-4):
+    ''' Apply BatchNorm, Relu 3x3, Conv2D, optional bottleneck block and dropout
```
Comment (kemaswill) [inline keras_contrib/applications/densenet.py]: What does ReLu 3x3 mean? Do you mean 'ReLU, Conv2D 3x3'?

File: keras_contrib/applications/densenet.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+    Args:
+        ip: Input keras tensor
+        nb_filter: number of filters
+        bottleneck: add bottleneck block
+        dropout_rate: dropout rate
+        weight_decay: weight decay factor
+
+    Returns: keras tensor with batch_norm, relu and convolution2d added (optional bottleneck)
+
```
Comment (kemaswill) [inline keras_contrib/applications/densenet.py]: Delete empty line here.

File: examples/densenet_cifar10.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+callbacks = [lr_reducer, early_stopper, model_checkpoint]
+
+model.fit_generator(generator.flow(trainX, Y_train, batch_size=batch_size), samples_per_epoch=len(trainX),
+                    nb_epoch=nb_epoch,
+                    callbacks=callbacks,
+                    validation_data=(testX, Y_test),
+                    nb_val_samples=testX.shape[0], verbose=2)
+
+yPreds = model.predict(testX)
```
Comment (kemaswill) [inline examples/densenet_cifar10.py]: Model should be evaluated using model.evaluate(X_test, Y_test).

**Comment by titu1994** [comment]
Corrected the scripts according to @kemaswill change request.

File: examples/cifar10_densenet.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+from keras.optimizers import Adam
+from keras.preprocessing.image import ImageDataGenerator
+from keras.utils import np_utils
+from keras_contrib.applications.densenet import DenseNet
+
+'''
+Trains a DenseNet-40-12 model on the CIFAR-10 Dataset.
+
+Gets a 99.84% accuracy score after 300 epochs.
+'''
```
Comment (kemaswill) [inline examples/cifar10_densenet.py]: Could you move these comments to the very beginning of this script?

Comment (titu1994) [inline examples/cifar10_densenet.py]: Done.

**Comment by kemaswill** [comment]
LGTM.

**Pull request merged**